### PR TITLE
Add minutes to hourly weather forecast time display

### PR DESCRIPTION
**Pull request by spacegaier**
## Proposed change

Might be subjective, but to me the hourly weather forecast looks weird. The numbers could easily indicate day of the month as well. So adding the minutes makes it clear and has no impact on the size, since we are not getting wider as the forecast icon already is.

**Before:**
image

image

**After:**
image

image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [X] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/dialogs/more-info/controls/more-info-weather.ts [hunk 1]
```diff
@@ -267,6 +267,7 @@ class MoreInfoWeather extends LitElement {
     return date.toLocaleDateString(this.hass.language, {
```
Comment (bramkragten) [inline src/dialogs/more-info/controls/more-info-weather.ts:267]: We should use <https://github.com/home-assistant/frontend/blob/dev/src/common/datetime/format_date.ts#L4>

File: src/panels/lovelace/cards/hui-weather-forecast-card.ts [hunk 2]
```diff
@@ -288,6 +288,7 @@ class HuiWeatherForecastCard extends LitElement implements LovelaceCard {
                                 this.hass!.language,
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-weather-forecast-card.ts:288]: https://github.com/home-assistant/frontend/blob/dev/src/common/datetime/format_time.ts#L4

File: src/dialogs/more-info/controls/more-info-weather.ts (hunk 1 above)
Comment (bramkragten) [reply src/dialogs/more-info/controls/more-info-weather.ts:267]: https://github.com/home-assistant/frontend/blob/dev/src/common/datetime/format_time.ts#L4

File: src/common/datetime/format_time.ts [hunk 3]
```diff
@@ -17,3 +17,12 @@ export const formatTimeWithSeconds = toLocaleTimeStringSupportsOptions
...
   : (dateObj: Date) => format(dateObj, "mediumTime");
+
+export const formatTimeWeekday = toLocaleTimeStringSupportsOptions
+  ? (dateObj: Date, locales: string) =>
+      dateObj.toLocaleTimeString(locales, {
+        weekday: "long",
+        hour: "numeric",
+        minute: "2-digit",
+      })
+  : (dateObj: Date) => format(dateObj, "shortTime");
```
Comment (bramkragten) [inline src/common/datetime/format_time.ts:28]: This is not correct, check the right format in the docs of fecha <https://github.com/taylorhakes/fecha#formatting-tokens>

File: src/common/datetime/format_date.ts [hunk 4]
```diff
@@ -9,3 +9,12 @@ export const formatDate = toLocaleDateStringSupportsOptions
...
   : (dateObj: Date) => format(dateObj, "longDate");
+
+export const formatDateWeekday = toLocaleDateStringSupportsOptions
+  ? (dateObj: Date, locales: string) =>
+      dateObj.toLocaleDateString(locales, {
+        weekday: "long",
+        month: "short",
+        day: "numeric",
+      })
+  : (dateObj: Date) => format(dateObj, "longDate");
```
Comment (bramkragten) [inline src/common/datetime/format_date.ts:20]: This is not correct, check the right format in the docs of fecha <https://github.com/taylorhakes/fecha#formatting-tokens>

File: src/common/datetime/format_time.ts (hunk 3 above)
Comment (spacegaier) [reply src/common/datetime/format_time.ts]: Not sure how I missed that 🤦 ... fixed now.

File: src/common/datetime/format_date.ts [hunk 5]
```diff
@@ -9,3 +9,12 @@ export const formatDate = toLocaleDateStringSupportsOptions
...
   : (dateObj: Date) => format(dateObj, "longDate");
+
+export const formatDateWeekday = toLocaleDateStringSupportsOptions
+  ? (dateObj: Date, locales: string) =>
+      dateObj.toLocaleDateString(locales, {
+        weekday: "long",
+        month: "short",
+        day: "numeric",
+      })
+  : (dateObj: Date) => format(dateObj, "dddd, D MMM");
```
Comment (bramkragten) [inline src/common/datetime/format_date.ts:20]:
```suggestion
  : (dateObj: Date) => format(dateObj, "dddd, MMM D");
```

**Review by bramkragten — APPROVED** [review]

**Pull request merged**